docs/RFCS/views.md

```diff
@@ -0,0 +1,143 @@
+Materialized views are explicitly out of scope.
+
+# Motivation
+
+[Views](https://en.wikipedia.org/wiki/View_(SQL)) are a widely-supported
+feature across
+[all](https://www.postgresql.org/docs/9.1/static/sql-createview.html)
+[major](http://dev.mysql.com/doc/refman/5.7/en/views.html)
+[SQL](https://msdn.microsoft.com/en-us/library/ms187956.aspx)
+[databases](http://www.ibm.com/support/knowledgecenter/SSEPEK_10.0.0/intro/src/tpc/db2z_views.html).
+In a sense, they're table stakes. Views are used for a number of reasons,
+including aliasing complex queries, limiting access to underlying data, or
+maintaining compatibility with legacy code as changes are made to the underlying
+database schema.
+
+# Scope
+
+As a bare minimum, we need to support creating views, referencing views
+in queries, and dropping views. We should probably also support altering
+views, although it would be possible to get good use out of views
+without that.
+
+Beyond the basics, though, different major SQL databases offer differing
+features around views. Some allow writing to underlying tables through views
+and checking the integrity of such updates. Some support a
+`CREATE OR REPLACE` statement to change a view's definition in a single
+command or idempotently create a view. Some have special restrictions on
+the `CREATE OR REPLACE` command. Some allow additional options on views,
+such as whether they're only temporary for the current session.
+
+Given our PostgreSQL compatibility, it makes sense to support what they
+support unless we have reason not to.
+
+* Even though it isn't part of the SQL standard,
+  [Postgres supports](https://www.postgresql.org/docs/9.1/static/sql-createview.html)
+  the `CREATE OR REPLACE` statement so long as the replacement query
+  outputs the same columns as the original query in the same order
+  (i.e. it can only add new columns to the end).
+* We should also support the applicable limited `ALTER VIEW` options that
+  [Postgres offers](https://www.postgresql.org/docs/9.1/static/sql-alterview.html).
+* Posgres supports views over
+  [VALUES statements](https://www.postgresql.org/docs/9.1/static/sql-values.html),
+  while MySQL does not. We have taken the approach of supporting VALUES
+  statements elsewhere, and will do so here as well.
+* Even though it is intentionally undefined by the SQL standard (and despite
+  [SQL Server's lack of support](https://msdn.microsoft.com/en-us/library/ms187956.aspx#Anchor_1),
+  both Postgres and MySQL support `ORDER BY` clauses on the select
+  statements used in view definitions, and respect them in later queries.
+  For compatibility reasons, we should try to do so as well.
+* We should support the `RESTRICT` option as the default on
+  [`DROP VIEW`](https://www.postgresql.org/docs/9.1/static/sql-dropview.html).
+  While supporting `CASCADE` as well would be nice, it can implemented
+  separately at a later time, as we
+  [chose to do with foreign keys](fk.md#cascade-and-other-behaviors).
```

Comment:
fwiw, FKs do support CASCADE for DROP (it just drops the FK constraint), just not for DELETE/TRUNCATE (where it'd need to find and delete actual rows)